### DISCUSSION TITLE
Unreachable assets

### DISCUSSION THREAD

**Discussion by loranger** [filamentphp/filament#460] [Help] [answered]
**Describe the bug**
Once deployed in production, js and css assets are unreachable and admin does not work, because `Uncaught ReferenceError: Livewire is not defined`

**To reproduce**
1. Develop a full admin locally
2. Deploy whole app (remote git clone, then composer install for php dependencies, front assets have already been compiled with `npm run prod` locally)
3. Admin does not work due to assets unreachable where clicking on unreachable url loads them perfectly
4. Browser console display the following errors :

```
GET .../admin/assets/css/filament.css?id=676bb703b5486e84c972 net::ERR_HTTP2_PROTOCOL_ERROR 200
login:233 
GET .../livewire/livewire.js?id=54d078b2ce39327a1702 net::ERR_HTTP2_PROTOCOL_ERROR 200
login:239 Uncaught ReferenceError: Livewire is not defined
    at login:239
(anonymous) @ login:239
login:270 
GET .../admin/assets/js/filament.js?id=848d169218fac889d48f net::ERR_HTTP2_PROTOCOL_ERROR 200
```

**Expected behavior**
Admin pages should work as locally, every other route respond, but not the admin ones.

**Screenshots**
![screen1](https://user-images.githubusercontent.com/6014/122084041-bad8ed80-ce01-11eb-8984-ea5354ee31a9.jpg)

![screen2](https://user-images.githubusercontent.com/6014/122084050-bc0a1a80-ce01-11eb-837a-20d350587789.png)

![screen3](https://user-images.githubusercontent.com/6014/122084053-bca2b100-ce01-11eb-9acc-83194e4056c0.jpg)

![screen4](https://user-images.githubusercontent.com/6014/122084058-bdd3de00-ce01-11eb-991f-df484f67a7d4.jpg)

**Context**
- Package version: 1.10
- Livewire version: dependency 2.4
- Laravel version: 8.45.1
- Browser: Firefox 91, Brave 1.25.72
- Server OS: Debian 10 prod + local
- PHP version: 7.4.16
- Database: MySQL 5.6.50 prod + local (docker)
- Web Server: Apache 2.4 prod + local (docker)

**Additional details**
I tried to `cache:clear` `route:clear` `config:cache` but nothing works.
I suspected assets url to be responsible of this weird behaviour, but it seems livewire.js is called relatively and filament.js is using absolute url. The result is the same: The files are not loaded and I can't figure out why

Did I miss something ?

**Comment by danharrin** [comment]
Hey... unfortunately I just can't seem to replicate this anywhere, which leads me to believe it's something to do with your hosting configuration... also Livewire doesn't seem to be working either and it runs on the same assets system as Filament does. If you have any luck reconfiguring your server and find a solution, let me know! Until then, I'm going to close this issue and move it to a discussion.

**Comment by leo-unglaub** [comment]
Your production webserver config is wrong. I had the same issue. Those assets do not really exist on the filesystem under this url. Because the `/admin` path is dynamic they are streamed via PHP. Your webserver is configures to only send `.php` files to the PHP interpterer, so your webserver looks for the .css and .js file in the filesystem. but this is a route. You have to add a rewrite rule to send `/admin/assets/*` also thru your PHP interpeter. Then it works fine!

**Reply by johnpuddephatt** [reply]
I had the following location block in my nginx config that was causing the problem described. Removing it meant Filament worked as expected.

```
location ~* ^.+\.(css|js|jpg|jpeg|gif|png|ico|gz|svg|svgz|ttf|otf|woff|eot|mp4|ogg|ogv|webm|webp|zip|swf)$ {
    add_header Access-Control-Allow-Origin "*";
    expires max;
    access_log off;
}
```

**Comment by loranger** [comment, marked as answer]
My client website is hosted on a shared host at OVH.
I finally made it work by adding the following rewrite rule at the bottom fo my laravel `.htaccess`

```apache
# Fix HTTP2 Bug on OVH
RewriteRule ^(.*)$ $1 [NS,E=no-gzip:1,E=dont-vary:1]
```

**Reply by Ariziky** [reply]
Hi, this worked for me. Thanks a lot !

**Reply by Iamkhaledkhan** [reply]
doesn't worked for me!

**Reply by gboquizosanchez** [reply]
Thank you so much, man. This was so annoying, and I didn't find anything about this problematic. So really thanks for this.
